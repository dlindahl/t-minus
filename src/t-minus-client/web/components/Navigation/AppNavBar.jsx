import { isString } from 'lodash'
import { Children, PropTypes } from 'react'

const baseStyles = {
  filler: {
    flex: 1
  },
  item: {
    flex: 1,
    textAlign: 'center'
  },
  items: {
    display: 'flex',
    height: '100%',
    listStyleType: 'none',
    margin: 0,
    padding: 0
  },
  root: {
    height: 100
  }
}
const NonBreakingSpace = 160
const StringStart = 0

function itemize (node) {
  let style = baseStyles.item
  if (isString(node) && node.charCodeAt(StringStart) === NonBreakingSpace) {
    style = baseStyles.filler
  }
  return (
    <li style={style}>
      {node}
    </li>
  )
}

const AppNavBar = (props) => {
  const items = Children.map(props.children, itemize)
  return (
    <nav style={baseStyles.root}>
      <ul style={baseStyles.items}>
        {items}
      </ul>
    </nav>
  )
}

AppNavBar.propTypes = {
  children: PropTypes.node
}

export default AppNavBar
