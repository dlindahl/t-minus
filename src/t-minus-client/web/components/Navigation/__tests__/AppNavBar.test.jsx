import AppNavBar from '../AppNavBar'
import renderer from 'react-test-renderer'

describe('AppNavBar', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <AppNavBar/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders itemized children', () => {
    const component = renderer.create(
      <AppNavBar>
        <div>TITLE</div>
        <button>BUTTON</button>
      </AppNavBar>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
