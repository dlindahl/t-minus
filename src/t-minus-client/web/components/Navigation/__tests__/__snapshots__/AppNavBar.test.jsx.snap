exports[`AppNavBar renders itemized children 1`] = `
<nav
  style={
    Object {
      "height": 100,
    }
  }>
  <ul
    style={
      Object {
        "display": "flex",
        "height": "100%",
        "listStyleType": "none",
        "margin": 0,
        "padding": 0,
      }
    }>
    <li
      style={
        Object {
          "flex": 1,
          "textAlign": "center",
        }
      }>
      <div>
        TITLE
      </div>
    </li>
    <li
      style={
        Object {
          "flex": 1,
          "textAlign": "center",
        }
      }>
      <button>
        BUTTON
      </button>
    </li>
  </ul>
</nav>
`;

exports[`AppNavBar renders the component 1`] = `
<nav
  style={
    Object {
      "height": 100,
    }
  }>
  <ul
    style={
      Object {
        "display": "flex",
        "height": "100%",
        "listStyleType": "none",
        "margin": 0,
        "padding": 0,
      }
    } />
</nav>
`;
